pkg/objstore/swift/swift.go

```diff
@@ -82,215 +206,141 @@ func (c *Container) Name() string {
 
 // Iter calls f for each entry in the given directory. The argument to f is the full
 // object name including the prefix of the inspected directory.
-func (c *Container) Iter(ctx context.Context, dir string, f func(string) error) error {
-	// Ensure the object name actually ends with a dir suffix. Otherwise we'll just iterate the
-	// object itself as one prefix item.
+func (c *Container) Iter(_ context.Context, dir string, f func(string) error) error {
 	if dir != "" {
-		dir = strings.TrimSuffix(dir, DirDelim) + DirDelim
+		dir = strings.TrimSuffix(dir, string(DirDelim)) + string(DirDelim)
 	}
-
-	options := &objects.ListOpts{Full: true, Prefix: dir, Delimiter: DirDelim}
-	return objects.List(c.client, c.name, options).EachPage(func(page pagination.Page) (bool, error) {
-		objectNames, err := objects.ExtractNames(page)
+	return c.connection.ObjectsWalk(c.name, &swift.ObjectsOpts{Prefix: dir, Delimiter: DirDelim}, func(opts *swift.ObjectsOpts) (interface{}, error) {
+		objects, err := c.connection.ObjectNames(c.name, opts)
 		if err != nil {
-			return false, err
+			return objects, errors.Wrap(err, "swift list object names")
 		}
-		for _, objectName := range objectNames {
-			if err := f(objectName); err != nil {
-				return false, err
+		for _, object := range objects {
+			if object == SegmentsDir {
```

Comment:
@bwplotka @GiedriusS PTAL this feels bit hacky, Swift puts the object segments to the `segments` directory in the configured segments container(bucket). To avoid breaking change and requiring another container, the default segments container is the same as for the original data. This means that in the bucket is directory `segments` full of the file parts. 

In one of the tests we check listing of all objects in the bucket, this is an issue in this case since there is additional object now causing the test to fail. As a solution I added skip of this one particular directory but I'm not sure it this is the right way and it feels bit hacky :/

Reply:
Ok, I'll leave it like it is